javalib/src/main/scala/java/lang/Long.scala

```diff
@@ -2,43 +2,62 @@ package java.lang
 
 import scalanative.runtime.{select, divULong, remULong, undefined, Intrinsics}
 
-final class Long(val longValue: scala.Long)
-    extends Number
-    with Comparable[Long] {
+final class Long(val _value: scala.Long) extends Number with Comparable[Long] {
   @inline def this(s: String) =
     this(Long.parseLong(s))
 
   @inline override def byteValue(): scala.Byte =
-    longValue.toByte
+    _value.toByte
 
   @inline override def shortValue(): scala.Short =
-    longValue.toShort
+    _value.toShort
 
-  @inline def intValue(): scala.Int =
-    longValue.toInt
+  @inline override def intValue(): scala.Int =
+    _value.toInt
 
-  @inline def floatValue(): scala.Float =
-    longValue.toFloat
+  @inline override def longValue(): scala.Long =
+    _value
 
-  @inline def doubleValue(): scala.Double =
-    longValue.toDouble
+  @inline override def floatValue(): scala.Float =
+    _value.toFloat
+
+  @inline override def doubleValue(): scala.Double =
+    _value.toDouble
 
   @inline override def equals(that: Any): scala.Boolean =
     that match {
       case that: Long =>
-        longValue == that.longValue
+        _value == that._value
       case _ =>
         false
     }
 
   @inline override def hashCode(): Int =
-    Long.hashCode(longValue)
+    Long.hashCode(_value)
 
   @inline override def compareTo(that: Long): Int =
-    Long.compare(longValue, that.longValue)
+    Long.compare(_value, that._value)
 
   @inline override def toString(): String =
-    Long.toString(longValue)
+    Long.toString(_value)
+
+  @inline override def __scala_==(other: _Object): scala.Boolean =
+    other match {
+      case other: java.lang.Byte    => _value == other._value.toLong
+      case other: java.lang.Short   => _value == other._value.toLong
+      case other: java.lang.Integer => _value == other._value.toLong
+      case other: java.lang.Long    => _value == other._value
+      case other: java.lang.Float   => _value.toFloat == other._value
+      case other: java.lang.Double  => _value.toDouble == other._value
+      case _                        => super.__scala_==(other)
```

Comment:
Shouldn't you put `case other: java.lang.Long` first as a fast-path? And similarly for all other variants.